bookserver/routers/rslogging.py

```diff
@@ -86,4 +107,75 @@ def set_tz_offset(
     values["tz_offset"] = tzreq.timezoneoffset
     response.set_cookie(key="RS_info", value=str(json.dumps(values)))
     rslogger.debug("setting timezone offset in session %s hours" % tzreq.timezoneoffset)
-    return "done"
+
+    return make_json_response()
+
+
+# runlog endpoint
+# ---------------
+# The `logRunEvent` client-side function calls this endpoint to record an activecode run
+@router.post("/runlog")
+async def runlog(request: Request, response: Response, data: LogRunIncoming):
+    # First add a useinfo entry for this run
+    if request.state.user:
+        if data.course != request.state.user.course_name:
+            return make_json_response(
+                status=status.HTTP_401_UNAUTHORIZED,
+                detail="You appear to have changed courses in another tab.  Please switch to this course",
```

Comment:
I agree that localization should be a thing.  Not anything we have done so far.  But perhaps this is something we should do as part of this porting process?

I feel like we need to be careful about how much we try to accomplish in this initial work of getting a book server up and running.  There is a tradeoff between getting a 1.0 of this up and running versus making it "perfect" for 1.0

Reply:
I agree. Let's KISS and get something working. Localization / nicer client-side handling of errors can come later. I'd suggesting building in server responses (short/simple) that make this possible in the future.

Reply:
Its a delicate balance between staying terse versus giving enough detail for a message to a user to be meaningful.

Reply:
I see these messages as internal only -- will they be reported directly to the user? I assume there's at least some GUI between the JSON and what the user sees anyway.

Reply:
Again, I'm mostly just porting over what we have and making it work. What we have done is just parrot these messages to the user with a simple alert in many cases.  Rethinking all of this is fine, I just want you to understand.

Reply:
I see. We can certainly put this in a todo list for later, then.

Reply:
I think it comes down to where we want to control what the user sees.  I thought it was reasonable to have that controlled on the server side.  And since this is all encapsulated in a json object now anyway I guess we can decide in the future.  The validation messages are a bit longer, and I forsee us just using the information from the detail to compose a message to the user.

I'm going to merge our two PRs now and we can continue.

I think with a couple more strategic APIs from ajax we'll have uncovered most of what we need to make this a really good first release.  That will give everyone a good example to follow as they contribute to the book server.